integration-tests/src/server.js

```diff
@@ -14,50 +13,43 @@ const { delay, withScreenshot } = require('../utils/integration-utils')
 const envs = require('../utils/terra-envs')
 
 
-// Sane-ish max runtime for a test. Mainly to extend the default 2-minute timeout.
+// Sane-ish max runtime for a test (20 minutes plus an extra minute for tear-down). Mainly to extend the default 2-minute timeout.
 const serverTimeout = 21 * 60 * 1000
 
 const app = express()
 const server = http.createServer(app)
-server.setTimeout(serverTimeout) // allow an extra minute for teardown
-
-const getBrowser = _.once(() => puppeteer.launch())
-const getContext = async () => {
-  const browser = await getBrowser()
-  return browser.createIncognitoBrowserContext()
-}
+server.setTimeout(serverTimeout)
 
 const testTimeout = async timeout => {
   await delay(timeout)
   throw new Error(`Test timeout after ${timeout}ms`)
 }
 
+const runTest = fn => withPuppeteer(async ({ browser, context, name, page, req, timeout }) => {
```

Comment:
I see what you're saying. My intent here was to separate concepts for readability, which I think aligns with your point #2. I really wanted to address this now to avoid the risk of not getting back around to it. However, at this point, I think I'll need to just merge and trust that my curiosity will bring me back around to this.